algorithms/a3c/main.py

```
@@ -40,80 +54,276 @@
 
 if __name__ == '__main__':
-    os.environ['OMP_NUM_THREADS'] = '1'
     os.environ['CUDA_VISIBLE_DEVICES'] = ""
 
+    # todo add to README.md more info
+
     args = parser.parse_args()
+    # set to 0 so that checkpoint resume can overwrite if necessary
+    args.episode_number = 0
+    args.total_length = 0
+    args.num_backprops = 0
+    if not args.experiment_id:
+        args.experiment_id = datetime.datetime.now().strftime("%Y-%m-%d-") \
+                                                                     + str(uuid.uuid4())
+    args.experiment_path = os.path.join(os.path.abspath(os.path.join(os.path.dirname(__file__),
+                           '..', '..', 'experiments', args.experiment_id)))
+    args.checkpoint_path = os.path.join(args.experiment_path, 'checkpoints')
+    args.tensorboard_path = os.path.join(args.experiment_path, 'tensorboard_logs')
 
-    torch.manual_seed(args.seed)
+    # Create environment from choices of Atari, ViZDoom and our ai2thor wrapper with tasks
     if args.atari:
         env = create_atari_env(args.atari_env_name)
-        args.frame_dim = 42  # fixed to be 42x42 in envs.py _process_frame42()
+        args.resolution = (42, 42)  # fixed to be 42x42 for _process_frame42() in envs.py
+    elif args.vizdoom:
+        # many more dependencies required for VizDoom so therefore import here
+        from algorithms.a3c.env_vizdoom import GroundingEnv
+
+        if args.evaluate == 0:
+            args.use_train_instructions = 1
+        elif args.evaluate == 1:
+            args.use_train_instructions = 1
+            args.num_processes = 0
+        elif args.evaluate == 2:
+            args.use_train_instructions = 0
+            args.num_processes = 0
+        else:
+            assert False, "Invalid evaluation type"
+
+        env = GroundingEnv(args)
+        env.game_init()
+        args.resolution = (args.frame_width, args.frame_height)
     else:
```

else should raise an exception for "environment not supported". Check always strings of env name even for ai2thor